src/lib/exec.rs

```diff
@@ -265,6 +280,28 @@ impl Executor for Interpreter {
                     LogOp::Or => to_value(v_a || v_b),
                 })
             }
+            ExprDef::BinOp(BinOp::Assign(ref op), ref a, ref b) => {
+                let v_r_b = self.run(b)?;
+                let v_b = (*v_r_b).clone();
+                let value;
+                match a.def {
+                    ExprDef::Local(ref name) => {
+                        let v_r_a = self.environment.get_binding_value(&name);
+                        let v_a = (*v_r_a).clone();
+                        value = exec_assign_op(op, v_a, v_b);
+                        self.environment
+                            .set_mutable_binding(&name, value.clone(), true);
+                    }
+                    ExprDef::GetConstField(ref obj, ref field) => {
+                        let v_r_a = self.run(obj)?;
```

Comment:
The spec says the left hand side of compound assignment is evaluated first. In `f().x += g()`, first `f` is called, then `g`.

Reply:
Thanks for pointing out, will have a look at it later.